tests/specs/controllers/FilesCtrlSpec.js

```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+describe('FilesCtrl', function() {
+    var scope = {}, routeParams = {}, rootScope = {};
+    routeParams.fileName = "/home/Desktop";
+    rootScope.ROOTHOME = "/home";
+    rootScope.FILES = {"/home/Desktop" : ["/home/Desktop/test.py"]};
+    rootScope.resultFiles = {"/home/Desktop/test.py" : [{"start": 2,
+                                      "end": 5,
+                                      "diffs": '+diffs',
+                                      "message": 'test',
+                                      "origin": 'PEP8',
+                                      "severity": 'NORMAL'
+                                    }]};
+    rootScope.FILE_DATA = {"/home/Desktop/test.py" : ["import this\n"]};
+
+    describe('parseCoalaProject()', function() {
+
+        beforeEach(module('coalaHtmlApp'));
+        beforeEach(inject(function ($controller) {
+            $controller('FilesCtrl', {
+                $scope: scope,
+                $routeParams: routeParams,
+                $rootScope: rootScope
+            });
+        }));
+
+        it('test fileName is set correctly', function() {
+            expect(scope.fileName).to.equal(routeParams.fileName);
+        });
+        it('test fileBack is obtained correctly', function() {
+            expect(scope.fileBack).to.equal('/home');
+        });
+        it('test fileType', function() {
+            expect(scope.fileType).to.equal('dir');
+        });
+        it('test fileContents', function() {
+            expect(scope.fileContents[0].name).to.equal('/home/Desktop/test.py');
+            expect(scope.fileContents[0].type).to.equal('file');
+            expect(scope.fileContents[0].result).to.true;
```

Comment:
It's better to compare the whole list, that you'll make sure that there is nothing redundant as well, something like:
```
expect(scope.fileContents).to.equal(...);
```